File: src/main/java/com/github/liebharc/queryenrichment/QueryFilter.java
package com.github.liebharc.queryenrichment;

import java.io.Serializable;

/**
 * A filter which is implemented by a {@link Query}. That is filters could also be executed during a {@link FilterStep}.
 */
public class QueryFilter {
    /** A filter expression */
    private final SimpleExpression filter;
    /** The query column for the filter */
    private final String column;

    public QueryFilter(SimpleExpression filter) {
        this(filter, filter.getAttribute().getProperty());
    }

    public QueryFilter(SimpleExpression filter, String column) {
        this.filter = filter;
        this.column = column;
    }

    public SimpleExpression getFilter() {
        return filter;
    }

    public String getColumn() {
        return column;
    }

    /**
     * Creates a SQL query expression with a placeholder value (e.g. "ID = ?").
     */
    public String toPlaceHolderString() {
        return column + filter.getOperation() + "?";
    }
}

File: src/main/java/com/github/liebharc/queryenrichment/QueryFilter.kt
package com.github.liebharc.queryenrichment

/**
 * A filter which is implemented by a [Query]. That is filters could also be executed during a [FilterStep].
 */
class QueryFilter constructor(
        /** A filter expression  */
        private val filter: SimpleExpression<*>,
        /** The query column for the filter  */
        val column: String = filter.attribute.property) {

    /**
     * Creates a SQL query expression with a placeholder value (e.g. "ID = ?").
     */
    fun toPlaceHolderString(): String {
        return column + filter.operation + "?"
    }
}
